appendices/VK_EXT_physical_device_drm.adoc

```diff
@@ -25,12 +25,19 @@ Unlike the EGL extension, this extension does not expose a string containing
 the name of the device file and instead exposes device minor numbers.
 
 DRM defines multiple device node types.
-Each physical device may have one primary node and one render node
-associated.
+Each physical device is associated with a single underlying DRM device, which
+may have one primary node and one render node.
+The primary node, in the context of this extension, refers to the capability of
+that device node to be used as a display subsystem.
+Note that this is not the same definition as a "DRM primary device node" on
+Linux, which may exist even for devices without a display subsystem.
+
 Physical devices may have no primary node (e.g. if the device does not have
 a display subsystem), may have no render node (e.g. if it is a software
 rendering engine), or may have neither (e.g. if it is a software rendering
 engine without a display subsystem).
+Physical devices which don't have a display subsystem implemented as part of
+the same underlying DRM device report that as hasPrimary=false.
```

Comment:
This is not what our implementation is doing right now. Is this what other drivers are doing currently?

Reply:
This is not what drivers are currently doing but what I meant to put explicitly from what was hinted in the next paragraph (`Physical devices may have no primary node (e.g. if the device does not have a display subsystem)`).

To my knowledge what drivers in Mesa are doing is:
- v3dv: uses the other display DRM device as the primary. This is what we would like to fix here and prevent other drivers from following and applications to rely on.
- panvk: uses the DRM primary (cardN) node of the same render device as the primary, because it technically exists and from the DRM terminology it is in fact the "DRM primary node" (but is not a "display subsystem").
- pvr: same as panvk (but still pending to upstream Mesa).

I guess what panvk and pvr do is also acceptable to standardize on while similarly achieving the same goal, if we want to settle on something based on existing implementations. If so we also need to change the text to explicitly say that and probably remove parts of the text which associate primary devices as being display subsystems.

Reply:
Neither panvk nor pvr are (yet) conformant, and require scary environment variables to unmask them, so I'm comfortable with changing their behaviour to match new wording.